### DISCUSSION TITLE
Extend Apply Method

### DISCUSSION THREAD

**Discussion by firstsaofan** [JasperFx/marten#2568] [Q&A]
Hi, 

I would like to ask if I can pass a delegate to Marten to enhance each Apply() method. 

This function is to set some common fields.

I hope this is Implicit to developers.

**Comment by oskardudycz** [comment]
@firstsaofan, could you expand on your use case, why do you need it?

**Comment by firstsaofan** [comment]
Thank you very much for replying to me. 
Sorry, I didn't describe it clearly. The method we are manually calling now. 
For example, the method name is SetCommonFields(); 
This method assigns values to some public fields, such as creator, creation time, modifier, version of json.data, and so on. 
We hope this is implicit for developers. 
Similar to the discussion in Gitter. As shown in the following :
![SetCommonFields](https://user-images.githubusercontent.com/52733945/230522604-4cd74a06-7fb5-412b-8612-b3df6d0c88df.PNG)
@oskardudycz

**Comment by jeremydmiller** [comment]
Sorry man, you're going to have to delegate to a shared method within each apply. Sounds like a decent enhancement later though. 

And I'm doing some housecleaning and moving this to a discussion
